macros/src/lib.rs

```diff
@@ -129,7 +131,9 @@ impl DerivedTS {
     /// impl ts_rs::TS for B { /* .. */ }
     /// ```
     fn generate_generic_types(&self) -> TokenStream {
-        let generics = self.generics.type_params().map(|ty| ty.ident.clone());
+        let generics = self.generics.type_params()
+            .filter(|ty| !self.concrete.contains_key(&ty.ident))
+            .map(|ty| ty.ident.clone());
```

Comment:
Then here, we just don't generate the dummy for that parameter.

Reply:
I see! That's how `concrete` sidesteps the trait problem then. Awesome